### PR TITLE
Make abs macro AVR specific. This fixes gcc >= 6.2 compile issues. 

### PR DESCRIPTION
**Pull request by 3djc**
Tested on gcc 6.2 and 4.9.2

PLEASE test on CLANG and Visual Studio

### CODE REVIEW

**Comment by projectkk2glider** [comment]
Looks good. I can't test.

**Comment by kilrah** [comment]
OK for me on windows

**Comment by 3djc** [comment]
Tested on Sierra CLang 8.0.0

**Comment by 3djc** [comment]
Size checks

**AVR-9X**
built in
   text	   data	    bss	    dec	    hex	filename
  51572	     16	   2873	  54461	   d4bd	firmware.elf

macro
   text	   data	    bss	    dec	    hex	filename
  51568	     16	   2873	  54457	   d4b9	firmware.elf

**ARM-X9E**
built in
   text	   data	    bss	    dec	    hex	filename
 460508	   2852	  52928	 516288	  7e0c0	firmware.elf

macro
   text	   data	    bss	    dec	    hex	filename
 460524	   2852	  52928	 516304	  7e0d0	firmware.elf

**ARM-Horus**
built in
   text	   data	    bss	    dec	    hex	filename
1143352	   7420	 842268	1993040	 1e6950	firmware.elf

macro
   text	   data	    bss	    dec	    hex	filename
1143352	   7420	 842268	1993040	 1e6950	firmware.elf

File: radio/src/opentx.h [hunk 1]
```diff
@@ -872,14 +872,14 @@ void doLoopCommonActions();
 
 #define BITMASK(bit) (1<<(bit))
 
-/// liefert Dimension eines Arrays
+/// retruns the number of elements of an array
```
Comment (bsongis) [inline radio/src/opentx.h]: typo!

File: radio/src/opentx.h [hunk 2]
```diff
@@ -872,14 +872,14 @@ void doLoopCommonActions();
...
 #define DIM(arr) (sizeof((arr))/sizeof((arr)[0]))
 
-/// liefert Betrag des Arguments
+/// AVR compîler does not optimize 'abs' well
+#if defined(__AVR_LIBC_MAJOR__)
```
Comment (bsongis) [inline radio/src/opentx.h]: Is it really needed?
I have a 4bytes flash saving if I remove the line on AVR (tested on 9X stock)

**Comment by bsongis** [comment]
Which version of avr-gcc are you using and which options did you choose for the 9X target?

**Comment by 3djc** [comment]
I use avr-gcc version 4.7.2 , which is also the one used by the docker builder, and use no option at all (cmake -DPCB=9X). I have 51584 when using Next (so macro used), and 51588 without the macro

**Comment by bsongis** [comment]
You are right about the 4 bytes, but I think it is better now

**Pull request merged**